File: WinForms/VB/GanttMainDemo/CodeExamples/CriticalPath.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.DXperience.Demos.CodeDemo
Imports DevExpress.DXperience.Demos.CodeDemo.Data
Imports DevExpress.LookAndFeel
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGantt
Imports DevExpress.Utils.Menu
Imports DevExpress.Utils
Imports DevExpress.Utils.Svg
Imports Examples
Imports DevExpress.XtraGantt.Scheduling

Namespace Examples

    <DevExpress.DXperience.Demos.CodeDemo.CodeExampleClass("Critical Path", "CriticalPath.cs")>
    Public Class Format

        <DevExpress.DXperience.Demos.CodeDemo.CodeExampleSetUp>
        Public Shared Function SetUp(ByVal sampleHost As DevExpress.XtraEditors.XtraUserControl) As Object()
            Dim ganttControl As DevExpress.XtraGantt.GanttControl = New DevExpress.XtraGantt.GanttControl()
            ganttControl.OptionsBehavior.Editable = False
            ganttControl.Dock = System.Windows.Forms.DockStyle.Fill
            ganttControl.Parent = sampleHost
            ganttControl.TreeListMappings.KeyFieldName = "Id"
            ganttControl.TreeListMappings.ParentFieldName = "ParentId"
            ganttControl.ChartMappings.TextFieldName = "Name"
            ganttControl.ChartMappings.StartDateFieldName = "StartTime"
            ganttControl.ChartMappings.FinishDateFieldName = "EndTime"
            ganttControl.ChartMappings.ProgressFieldName = "Progress"
            ganttControl.ChartMappings.PredecessorsFieldName = "PredecessorIDs"
            ganttControl.DataSource = DevExpress.DXperience.Demos.CodeDemo.Data.Task.GetData()
            For i As Integer = 0 To 7 - 1
                ganttControl.WorkWeek.Add(New DevExpress.XtraGantt.WorkDayOfWeek(CType(i, System.DayOfWeek), New System.Collections.Generic.List(Of DevExpress.XtraGantt.Scheduling.WorkTime)() From {New DevExpress.XtraGantt.Scheduling.WorkTime(System.TimeSpan.FromHours(0), System.TimeSpan.FromHours(24))}))
            Next

            AddHandler ganttControl.Load, Sub(s, e) ganttControl.ExpandAll()
            Return New Object() {ganttControl}
        End Function

        <DevExpress.DXperience.Demos.CodeDemo.CodeExampleCase("Show a critical path")>
        <DevExpress.DXperience.Demos.CodeDemo.CodeExampleUnderlineTokens("OptionsView", "CriticalPathTask", "CriticalPathDependency")>
        Public Shared Sub ShowCriticalPath(ByVal ganttControl As DevExpress.XtraGantt.GanttControl)
            ganttControl.OptionsView.CriticalPathHighlightMode = DevExpress.XtraGantt.CriticalPathHighlightMode.[Single]
            ganttControl.Appearance.CriticalPathTask.BackColor = DevExpress.LookAndFeel.DXSkinColors.FillColors.Danger
            ganttControl.Appearance.CriticalPathDependency.BackColor = DevExpress.LookAndFeel.DXSkinColors.FillColors.Danger
        End Sub

        <DevExpress.DXperience.Demos.CodeDemo.CodeExampleCase("Show multiple critical paths")>
        <DevExpress.DXperience.Demos.CodeDemo.CodeExampleUnderlineTokens("OptionsView", "CriticalPathTask", "CriticalPathDependency")>
        Public Shared Sub ShowMultipleCriticalPaths(ByVal ganttControl As DevExpress.XtraGantt.GanttControl)
            ganttControl.OptionsView.CriticalPathHighlightMode = DevExpress.XtraGantt.CriticalPathHighlightMode.Multiple
            ganttControl.Appearance.CriticalPathTask.BackColor = DevExpress.LookAndFeel.DXSkinColors.FillColors.Danger
            ganttControl.Appearance.CriticalPathDependency.BackColor = DevExpress.LookAndFeel.DXSkinColors.FillColors.Danger
        End Sub

        <DevExpress.DXperience.Demos.CodeDemo.CodeExampleTearDown>
        Public Shared Sub TearDown(ByVal sampleHost As DevExpress.XtraEditors.XtraUserControl)
            For Each control As System.Windows.Forms.Control In sampleHost.Controls
                control.Dispose()
            Next

            sampleHost.Controls.Clear()
        End Sub
    End Class
End Namespace

File: WinForms/CS/GanttMainDemo/CodeExamples/CriticalPath.cs
namespace Examples {
    using System;
    using System.Collections.Generic;
    using System.Drawing;
    using System.Windows.Forms;
    using DevExpress.DXperience.Demos.CodeDemo;
    using DevExpress.DXperience.Demos.CodeDemo.Data;
    using DevExpress.LookAndFeel;
    using DevExpress.XtraEditors;
    using DevExpress.XtraGantt;
    using DevExpress.Utils.Menu;
    using DevExpress.Utils;
    using DevExpress.Utils.Svg;
    using Examples;
    using DevExpress.XtraGantt.Scheduling;

    [CodeExampleClass("Critical Path", "CriticalPath.cs")]
    public class Format {
        [CodeExampleSetUp]
        public static object[] SetUp(XtraUserControl sampleHost) {
            GanttControl ganttControl = new GanttControl();
            ganttControl.OptionsBehavior.Editable = false;
            ganttControl.Dock = DockStyle.Fill;
            ganttControl.Parent = sampleHost;
            ganttControl.TreeListMappings.KeyFieldName = "Id";
            ganttControl.TreeListMappings.ParentFieldName = "ParentId";
            ganttControl.ChartMappings.TextFieldName = "Name";
            ganttControl.ChartMappings.StartDateFieldName = "StartTime";
            ganttControl.ChartMappings.FinishDateFieldName = "EndTime";
            ganttControl.ChartMappings.ProgressFieldName = "Progress";
            ganttControl.ChartMappings.PredecessorsFieldName = "PredecessorIDs";
            ganttControl.DataSource = Task.GetData();
            for(int i = 0; i < 7; i++)
                ganttControl.WorkWeek.Add(new WorkDayOfWeek((DayOfWeek) i, new List<WorkTime>() {new WorkTime(TimeSpan.FromHours(0), TimeSpan.FromHours(24))}));
            ganttControl.Load += (s, e) => ganttControl.ExpandAll();
            return new object[] {ganttControl};
        }
        [CodeExampleCase("Show a critical path")]
        [CodeExampleUnderlineTokens("OptionsView", "CriticalPathTask", "CriticalPathDependency")]
        public static void ShowCriticalPath(GanttControl ganttControl) {
            ganttControl.OptionsView.CriticalPathHighlightMode = CriticalPathHighlightMode.Single;
            ganttControl.Appearance.CriticalPathTask.BackColor = DXSkinColors.FillColors.Danger;
            ganttControl.Appearance.CriticalPathDependency.BackColor = DXSkinColors.FillColors.Danger;
        }
        [CodeExampleCase("Show multiple critical paths")]
        [CodeExampleUnderlineTokens("OptionsView", "CriticalPathTask", "CriticalPathDependency")]
        public static void ShowMultipleCriticalPaths(GanttControl ganttControl) {
            ganttControl.OptionsView.CriticalPathHighlightMode = CriticalPathHighlightMode.Multiple;
            ganttControl.Appearance.CriticalPathTask.BackColor = DXSkinColors.FillColors.Danger;
            ganttControl.Appearance.CriticalPathDependency.BackColor = DXSkinColors.FillColors.Danger;
        }
        [CodeExampleTearDown]
        public static void TearDown(XtraUserControl sampleHost) {
            foreach(Control control in sampleHost.Controls)
                control.Dispose();
            sampleHost.Controls.Clear();
        }
    }
}
